### ISSUE TITLE
arm builds fail after move to wasm for front end

### ISSUE THREAD

**Issue by mcrosson** [Plume-org/Plume#394]
Plume version (and/or commit): 3fce5d6a14831745115ae9cc8d25ace2b6d70ec8

After version 3fce5d6a14831745115ae9cc8d25ace2b6d70ec8 Plume fails to build on arm with an error stating ```rust-lld not found```. It looks like wasm isn't building artifacts for arm and a few related tickets are below.

- https://github.com/rustwasm/team/issues/215
- https://github.com/rust-lang/rust/issues/52362

From what I can tell rustup may need some additional components on arm builds that aren't installed by default but I'm not 100% how to progress.

**Comment by marek-lach** [comment]
You might find this of use (or not): https://www.reddit.com/r/rust/comments/9a7te2/nightly_rust_is_switching_to_use_lld_llvms_new/

**Comment by mcrosson** [comment]
@marek-lach Thanks!

**Comment by marek-lach** [comment]
More specifically here: https://rust-embedded.github.io/blog/2018-08-2x-psa-cortex-m-breakage/

**Comment by mcrosson** [comment]
@marek-lach Thank you again 👍 

Now to figure out what the necessary linker=<value> pair needs to be on these 2 arm boards (one is arm32 with hardware float, the other arm64)

**Comment by mcrosson** [comment]
@marek-lach Might you have insights into the below error I just got?

```
root@d01464ecc387:/app# export PLUME_FRONT_RUST_FLAGS="-C linker=aarch64-linux-gnu-ld" 
root@d01464ecc387:/app# RUSTFLAGS="${PLUME_FRONT_RUST_FLAGS}" cargo web deploy --package plume-front
```

```
  = note: /app/target/wasm32-unknown-unknown/debug/deps/plume_front.14gnp5g9ev883vyh.rcgu.o: 
file not recognized: File format not recognized
```

**Comment by mcrosson** [comment]
Current progress. A shell script to build plume-front but it's currently failing with the following error. Shell script included after the error.

```

  = note: lld: error: unknown argument: --export-dynamic


error: aborting due to previous error
error: Could not compile `plume-front`.
```

```
#!/bin/sh

cd /opt

wget http://releases.llvm.org/7.0.0/clang+llvm-7.0.0-armv7a-linux-gnueabihf.tar.xz
wget http://releases.llvm.org/7.0.0/clang+llvm-7.0.0-aarch64-linux-gnu.tar.xz

tar -xf clang*
rm *.xz
mv clang*/* .
cd /app
export PATH_OLD=${PATH}
export PATH=/opt/bin:${PATH}
RUSTFLAGS="-C linker=lld" cargo web deploy -p plume-front
rm -rf /opt/*
export PATH=${PATH_OLD}
```

**Comment by marek-lach** [comment]
I am not sure, @BaptisteGelez might know more about the Rust frontend in Plume, but the lld error you've got looks somewhat similar to this? https://github.com/facebook/osquery/issues/3408#issuecomment-309233430

**Comment by fdb-hiroshima** [comment]
Fixed by #402
